roles/big-service/tasks/main.yml

```diff
@@ -0,0 +1,21 @@
+---
+
+- name: stopping the {{service_name}}
+  service: name={{service_name}} pattern=/app/{{service_name}}/{{script_name}} state=stopped
+  ignore_errors: True
+
+- name: delete old service directory
+  shell: rm -rf /app/{{service_name}}
```

Comment:
Have you tried something like `- file: path=/app/{{ service_name}} state=absent`?
Usually the usage of `shell` is discouraged (but I know there have been some issues with removal of directories)